### ISSUE TITLE
Add `from_unixtime(unixtime, precision)` convenience operation for sub-seconds since epoch

### ISSUE THREAD

**Issue by alec-heif** [prestosql/presto#5038]
Currently `from_unixtime` always takes in the number of seconds since the unix epoch. It would be nice to have a way to parse milliseconds, nanoseconds, etc.

Note that this might have to be implemented as a separate function since it appears there already exists a `from_unixtime(unixtime, zone)` and I suspect it will be too tricky to have two different overloads with the same number of arguments but different types.

**Comment by electrum** [comment]
`from_unixtime` actually takes the number of fractional seconds as a `double` and returns a `timestamp(3)`. Using a floating point number in this context is not ideal, and I don't think we should repeat this mistake in new variants.

The largest epoch unit for `timestamp` that fits in `bigint` is microseconds (the `ShortTimestampType` internal representation). If we wanted to accept nanoseconds, we'd either need to use a `decimal` (of the `LongDecimalType` variety) or split up the arguments into seconds and fractional seconds.

**Comment by alec-heif** [comment]
> the largest epoch unit for timestsamp that fits in `bigint` is microseconds

I agree this is true in the general case of "all possible times ever", but I did a quick back of the envelope check:
- `1598902433` - current unixtime
- `1598902433000000000` - current unixtime in nanos
- `9223372036854775808` - 2^63

shouldn't we have at least a couple hundred years worth of headroom before we're actually **forced** to use a decimal?

**Comment by alec-heif** [comment]
> Using a floating point number in this context is not ideal, and I don't think we should repeat this mistake in new variants.

I agree. Based on the feedback in #5037 it seems that the API here actually looks like four different functions `from_unixtime_(seconds)|(millis)|(micros)|(nanos)` that each solely take in a BIGINT number of <time unit> since the unix epoch.

**Comment by electrum** [comment]
That’s a good point. We can use bigint for common values and have a decimal overload for larger values (as we’d want to allow the function to work for any value the type supports).
